src/circuit/circuit_frame.rs

```diff
@@ -1327,7 +1327,11 @@ fn reduce_cons<F: LurkField, CS: ConstraintSystem<F>>(
 
     macro_rules! def_head_sym {
         ($var:ident, $field:ident) => {
-            let $var = head.alloc_equal(&mut cs.namespace(|| stringify!($var)), &g.$field)?;
+            let $var = alloc_equal(
+                &mut cs.namespace(|| stringify!($var)),
+                head.hash(),
+                &g.$field.hash(),
+            )?;
```

Comment:
Unfortunately, this removes the sym test but never fully replaces it (even the 'one time') needed as noted in the comment.

I pushed a test showing that the new code can produce a runtime error if a carefully-chosen value is substituted for a known symbol.

Reply:
Since I would like to try also refactoring to use constants rather than allocated values, I propose to take this over. I can fix the above problem as part of the same work.

Reply:
Ok, makes sense.

Reply:
I created a new PR (#258) that now makes the failing test pass. Since I had to rebase, and since it will streamline review, it was easier to make a new PR. Let's close this one in favor of that.